### PR TITLE
Allow user to reset their password

### LINKED ISSUE #112: Create forget password flow

### PR DESCRIPTION
**Pull request by fkoulen**
* Closes #112 
* Adds [tailwindcss-animated](https://www.tailwindcss-animated.com/) plugin
* Creates CSS classes for the styling used on 'auth pages' (sign-in, change-password etc.)
* Create pages for the reset password flow

### Screenshots 
image

### CODE REVIEW

File: src/components/forgot-password/forgot-password-form.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  return (
+    <div className={"auth-container text-center"} data-cy={"request-link"}>
+      <Image
+        src={"/images/undraw-forgot-password.svg"}
+        alt={"Person looking confused at password field"}
+        width={0}
+        height={0}
+        style={{ width: "100%" }}
```
Comment (rensdoodeman) [inline src/components/forgot-password/forgot-password-form.tsx:37]: why the width 0 and then 100%?

File: src/components/forgot-password/forgot-password-form.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      />
+      <span className={"text-xl font-semibold"}>Forgotten your password?</span>
+      <p className={"text-zinc-400"}>
+        Enter your email and we{`'`}ll send you a link to reset your password.
```
Comment (rensdoodeman) [inline src/components/forgot-password/forgot-password-form.tsx:41]:
```suggestion
        {"Enter your email and we'll send you a link to reset your password."}
```
```suggestion
        {constName}
```

File: src/services/authentication.service.ts [hunk 3]
```diff
@@ -12,3 +13,15 @@ export const setName = async (
...
 }
+
+export const resetPasswordForEmail = async (supabase: SupabaseClient, email: string) => {
+  const { error } = await supabase.auth.resetPasswordForEmail(email, {
+    redirectTo: `${window.location.origin}/api/auth/callback?next=/forgot-password/reset&`,
+  })
+  if (error) {
+    toast.error(error.message)
+    return false
+  } else {
```
Comment (rensdoodeman) [inline src/services/authentication.service.ts:24]: you technically don't need to wrap the `return true` in the else

**Review by rensdoodeman — APPROVED** [review]

File: src/components/forgot-password/forgot-password-form.tsx (hunk 1 above)
Comment (fkoulen) [reply src/components/forgot-password/forgot-password-form.tsx:37]: https://github.com/vercel/next.js/discussions/18474#discussioncomment-5501724

File: src/components/forgot-password/forgot-password-form.tsx (hunk 2 above)
Comment (fkoulen) [reply src/components/forgot-password/forgot-password-form.tsx:41]: constName?

File: src/services/authentication.service.ts (hunk 3 above)
Comment (fkoulen) [reply src/services/authentication.service.ts:24]: valid

**Pull request merged**